docs/src/components/hardware-list.tsx

```diff
@@ -0,0 +1,156 @@
+import React from "react";
+
+import {
+  Board,
+  HardwareMetadata,
+  Interconnect,
+  Shield,
+} from "../hardware-metadata";
+
+interface HardwareListProps {
+  items: HardwareMetadata[];
+}
+
+function itemIds(item: HardwareMetadata) {
+  if (item.type == "board" || item.type == "shield") {
+    let nodes = (item.siblings || [item.id])
+      .map<JSX.Element | string>((id) => <code key={id}>{id}</code>)
+      .reduce(
+        (prev, curr, index) => [...prev, index > 0 ? " and " : "", curr],
+        [] as Array<string | JSX.Element>
+      );
+    return <span key={item.id}>{nodes}</span>;
+  } else {
+    return <code key={item.id}>{item.id}</code>;
+  }
+}
+
+function typeLabel(type: HardwareMetadata["type"]): string {
+  switch (type) {
+    case "board":
+      return "Board: ";
+    case "shield":
+      return "Shield: ";
+    case "interconnect":
+      return "Interconnect: ";
+  }
+}
+
+function HardwareLineItem({ item }: { item: HardwareMetadata }) {
+  return (
+    <li>
+      <a href={item.url}>{item.name}</a> ({typeLabel(item.type)} {itemIds(item)}
+      )
+    </li>
+  );
+}
+
+function mapInterconnect({
+  interconnect,
+  boards,
+  shields,
+}: {
+  interconnect?: Interconnect;
+  boards: Array<Board>;
+  shields: Array<Shield>;
+}) {
+  if (!interconnect) {
+    return null;
+  }
+
+  return (
+    <div key={interconnect.id}>
+      <h4>{interconnect.name} Keyboards</h4>
+      {interconnect.description && <p>{interconnect.description}</p>}
+      <h5>Boards</h5>
+      <ul>
+        {boards.map((s) => (
+          <HardwareLineItem key={s.id} item={s} />
+        ))}
+      </ul>
+      <h5>Shields</h5>
+      <ul>
+        {shields.map((s) => (
+          <HardwareLineItem key={s.id} item={s} />
+        ))}
+      </ul>
+    </div>
+  );
+}
+
+function HardwareList({ items }: HardwareListProps) {
+  let grouped: {
+    onboard: Array<Board>;
+    interconnects: {
+      [key: string]: {
+        interconnect?: Interconnect;
+        boards: Array<Board>;
+        shields: Array<Shield>;
+      };
+    };
+  } = items.reduce(
+    (agg, hm) => {
+      switch (hm.type) {
+        case "board":
+          if (hm.features?.includes("keys")) {
+            agg.onboard.push(hm);
+          } else if (hm.exposes) {
+            hm.exposes.forEach((element) => {
+              let ic = agg.interconnects[element] || {
+                boards: [],
+                shields: [],
+              };
+              ic.boards.push(hm);
+              agg.interconnects[element] = ic;
```

Comment:
Thinking a bit about requires/exposes... Can't boards expose something like `i2c_oled`? This seems like a case where we may need to differentiate between interconnect requires/exposes and feature requires/exposes. In this case, `i2c_oled` isn't an interconnect, so this shouldn't be valid.

Reply:
But it *is* an interconnect, just not one that a shields with the `keys` feature would likely use.

I think the interconnect is truly generic, and what features show up with the selected shield you add to it drives the "did we get what we need?"

Reply:
Further example... That interconnect gets hijacked for pointers by modders...

Reply:
I see what you're saying, and I think it makes sense. The trouble I was having with this was during my build workflow rewrite, where I am compiling a similar list of all boards, shields, and interconnects for the nightly builds. The issue that crossed my mind was that I can't build the `i2c_oled` interconnect with a board and shield, but I think this is remedied by only building interconnects that have boards exposing it and shields requiring it. This would be a board + shield combo interconnect to build. Does that logic actually hold up?

Reply:
Only thing I would add is building those combinations that resulting in the final set of `features` including the `keys` feature. Otherwise currently ZMK fails without a kscan chosen.

Reply:
That additional check isn't needed *today* but might be needed later as we expand the use of interconnects more.